tabbycat/tournaments/forms.py

```diff
@@ -110,17 +116,19 @@ def save(self):
         # Identify + apply selected preset
         selected_index = self.cleaned_data["preset_rules"]
         selected_preset = next(p for p in presets if p.name == selected_index)
-        selected_preferences = get_preferences_data(selected_preset, t)
-        for preference in selected_preferences:
-            t.preferences[preference['key']] = preference['new_value']
+        save_presets(t, selected_preset)
 
         # Apply public info presets
         do_public = self.cleaned_data["public_info"]
-        public_preset = next((p for p in presets if p.name == do_public), False)
-        if public_preset:
-            public_preferences = get_preferences_data(public_preset, t)
-            for preference in public_preferences:
-                t.preferences[preference['key']] = preference['new_value']
+        if do_public:
+            save_presets(t, PublicInformation)
+
+        # Apply data entry method preset
+        data_entry_method = self.cleaned_data["data_entry"]
+        if data_entry_method == "private-urls":
+            save_presets(t, PrivateURLS)
+        elif data_entry_method == "public":
+            save_presets(t, PublicForms)
```

Comment:
```suggestion
        if data_entry_method:
            save_presets(t, {"private-urls": PrivateURLS, "public": PublicForms}[data_entry_method])
```
Easier to add new keys than to make a new `if`.